sdk/communication/azure-communication-administration/azure/communication/administration/aio/_polling_async.py

```diff
@@ -0,0 +1,95 @@
+# pylint: disable=W0231
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import asyncio
+from typing import Union
+import base64
+from functools import partial
+
+from azure.core.polling import AsyncPollingMethod
+
+from .._phonenumber._generated.models import (
+    PhoneNumberSearch,
+    PhoneNumberRelease
+)
+
+class PhoneNumberBasePollingAsync(AsyncPollingMethod):
+    """ABC class for reserve/purchase/release phone number related polling.
+    """
+    def __init__(self, is_terminated, interval=5):
+        self._response = None
+        self._client = None
+        self._query_status = None
+        self._is_terminated = is_terminated
+        self._polling_interval = interval
+
+    async def _update_status(self):
+        # type: () -> None
```

Comment:
can we use py3 only syntax here, since async code is only supported in py3, @xiangyan99

Reply:
This is in aio folder. Only aio folder contains async code.

Reply:
@turalf Yes. We can use py3 only syntax in async code. Async already implies py3+ :)